_ingest-pipelines/processors/ml-inference.md

```diff
@@ -166,20 +194,108 @@ The response confirms that, in addition to the `passage_text` field, the process
 Once you have created an ingest pipeline, you need to create an index for ingestion and ingest documents into the index.
 {: .note}
 
-### Example ingest processor configuration for local models
+### Example: Local model
 
-The following example creates an ingest pipeline for a `SENTENCE_TRANSFORMER` local model that requires an `text_docs` field, produces a `data` field for prediction in the model output path `$.inference_results.*.output.*.data`. The processor converts the `context` field into text embeddings and stores in the `context_embedding` field. The `input_map` and `output_map` mappings support standard JSON path notation for complex data structures. The model used here is a [pretrained model](https://opensearch.org/docs/latest/ml-commons-plugin/pretrained-models/#sentence-transformers) from ml-commons with the name `huggingface/sentence-transformers/all-distilroberta-v1`
+The following example configures an `ml_inference` processor with a local model.
 
-For local models, the `model_input` field is mandatory and you should specify the inputs and their formats you want to provide the ML Model. If there are any inputs in `model_config` map to be used by model, they should also be added to `model_input`
+**Step 1: Create a pipeline**
 
-For remote models, the `model_input` field is optional and its default value is `"{ \"parameters\": ${ml_inference.parameters} }`.
-{: .note}
+The following example creates an ingest pipeline for the `huggingface/sentence-transformers/all-distilroberta-v1` local model. The model is a sentence transformer [pretrained model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/pretrained-models/#sentence-transformers) hosted in your OpenSearch cluster. 
+
+If you invoke the model using the Predict API, the request looks like this:
+
+```json
+POST /_plugins/_ml/_predict/text_embedding/cleMb4kBJ1eYAeTMFFg4
+{
+  "text_docs":[ "today is sunny"],
+  "return_number": true,
+  "target_response": ["sentence_embedding"]
+}
+```
+
+Using this schema, you'll specify the `model_input` as follows:
+
+```json
+ "model_input": "{ \"text_docs\": ${input_map.text_docs}, \"return_number\": ${model_config.return_number}, \"target_response\": ${model_config.target_response} }"
+```
+
+The model converts the text in the `text_docs` field to embeddings so it expects an ingested document to contain a `text_docs` field. In this example, ingested documents do not contain a `text_docs` field. The `model_input` specifies to source the `text_docs` field from the `input_map`. The `input_map`, in turn, specifies that the input text resides at the `book.*.chunk.text.*.context` JSON path:
```

Comment:
Could you help simplify this part? We just say that we map `book.*.chunk.text.*.context` to model's input `text_docs` field. More details could be confusing sometime.